dashboard/app/views/layouts/application.html.haml

```diff
@@ -36,6 +35,16 @@
     - if current_user && current_user.age.nil?
       = render partial: 'layouts/age_interstitial'
 
+    #gdpr-dialog
+    - gdpr_data = {}
+    - if current_user
+      - already_accepted = !current_user.data_transfer_agreement_accepted.nil?
+      - force_eu = !request.params['force_in_eu'].nil?
+      - gdpr_data[:show_gdpr_dialog] = (request.gdpr? || force_eu) && !already_accepted
+      - gdpr_data[:current_user_id] = current_user.id
+
+    %script{src: minifiable_asset_path('js/code-studio.js'), data: {gdpr: gdpr_data.to_json}}
```

Comment:
Future (post May 25, even) refactoring thought: Can we extract all the GDPRDialog code to its own JavaScript entry point (out of code-studio.js) and not even render this section of application.html.haml if the user has already accepted the agreement?  Or does that mess with our caching setup somehow?